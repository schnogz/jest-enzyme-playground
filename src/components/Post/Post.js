import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
	height:20%;
	width: 16%;
	display:inline-block;
	margin:1%;
	border: 1px solid grey;
`;

const Title = styled.a`
  font-size: 1.2em;
  color: palevioletred;
  text-decoration: none;
`;

export default class Post extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<Wrapper>
				<Title href={this.props.post.url || '#'} target={'_blank'}>
					{this.props.post.title}
				</Title>
			</Wrapper>
		);
	}
};
