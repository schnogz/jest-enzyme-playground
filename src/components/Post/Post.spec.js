import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

import Post from './Post';

describe('Post component', () => {
	it('tree renders correctly with title', () => {
		const component = shallow(<Post post={{ title: 'Test', url: 'fake' }}/>);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});
});