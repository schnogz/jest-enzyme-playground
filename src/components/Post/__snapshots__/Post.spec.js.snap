// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component tree renders correctly with title 1`] = `
<styled.div>
  <styled.a
    href="fake"
    target="_blank"
  >
    Test
  </styled.a>
</styled.div>
`;
